pkg/capi/edit/ClusterConfig.vue

```diff
@@ -227,6 +247,14 @@ export default (Vue as VueConstructor<
       this.set(this.value.metadata, 'namespace', clusterClassNs);
     },
     async saveOverride() {
+      // These two fields have to be numbers
+      if (this.value?.spec?.controlPlaneEndpoint?.port) {
+        set(this.value.spec.controlPlaneEndpoint, 'port', Number(this.value.spec.controlPlaneEndpoint.port));
+      }
+      if (this.value?.spec?.clusterNetwork?.apiServerPort) {
+        set(this.value.spec.clusterNetwork, 'apiServerPort', Number(this.value.spec.clusterNetwork.apiServerPort));
+      }
```

Comment:
I would coerce these into numbers when they're changed rather than just before save. You can use `v-model.number` on `controlPlaneEndpoint.port` and coerce `clusterNetwork.apiServerPort` in the event handler eg `"(val) => $set(value.spec.clusterNetwork, 'apiServerPort', Number(val) || '')"` . The benefit of doing so is that they'll be numbers if you switch to the yaml view.